app/experimenter/experiments/models.py

```diff
@@ -804,6 +807,60 @@ def json_dumps_value(self):
         return json.dumps(json.loads(self.value), indent=2)
 
 
+class VariantPreferences(models.Model):
+    variant = models.ForeignKey(
+        ExperimentVariant,
+        blank=False,
+        null=False,
+        related_name="preferences",
+        on_delete=models.CASCADE,
+    )
+    pref_name = models.CharField(max_length=255, blank=False, null=False)
+    pref_type = models.CharField(
+        max_length=255,
+        choices=ExperimentConstants.PREF_TYPE_CHOICES,
+        blank=True,
+        null=True,
+    )
+    pref_branch = models.CharField(
+        max_length=255,
+        choices=ExperimentConstants.PREF_BRANCH_CHOICES,
+        blank=True,
+        null=True,
+    )
+
+    pref_value = models.CharField(max_length=255, blank=False, null=False)
```

Comment:
How come type and branch have blank/null=True but name/value don't?

Reply:
lemme fix this...